prime-router/src/main/kotlin/cli/tests/TestReportStream.kt

```diff
@@ -453,21 +453,23 @@ abstract class CoolTest {
         val hl7Sender = settings.findSender("$orgName.$hl7SenderName")
             ?: error("Unable to find sender $hl7SenderName for organization ${org.name}")
 
-        val allGoodReceivers = settings.receivers.filter {
-            it.organizationName == orgName &&
-                !it.name.contains("FAIL") &&
-                !it.name.contains("BLOBSTORE") &&
-                !it.name.contains("QUALITY") &&
-                !it.name.contains("AS2") &&
-                !it.name.contains("OTC")
-        }
-        val allGoodCounties = allGoodReceivers.map { it.name }.joinToString(",")
-
-        val csvReceiver = allGoodReceivers.filter { it.name == "CSV" }[0]
-        val hl7Receiver = allGoodReceivers.filter { it.name == "HL7" }[0]
-        val hl7BatchReceiver = allGoodReceivers.filter { it.name == "HL7_BATCH" }[0]
-        val redoxReceiver = allGoodReceivers.filter { it.name == "REDOX" }[0]
-        val hl7NullReceiver = allGoodReceivers.filter { it.name == "HL7_NULL" }[0]
+        val csvReceiver = settings.receivers.filter { it.organizationName == orgName && it.name == "CSV" }[0]
```

Comment:
Should this make sure that there is at least 1 result in the filtered sequence, since we're doing `[0]`? Maybe something like (make the message better though):
```kotlin
something.filter{it.name == "selector"}.elementAtOrElse(0) { index -> assert(false, "Element expected, but not found") }
```
It can assert or log, but _something_ that makes it blatantly clear from the output/log of "this is why this test failed" to make it easy to debug/remediate it.

Same for the lines below

Reply:
Yes, if this were not test code, I would definitely do this.